File: sdks/python/tests/test_live.py
"""Live integration test — requires bridge running on localhost:5000."""
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))

from antigravity_automation import AntigravityClient

client = AntigravityClient()
print("Testing against http://localhost:5000...\n")

try:
    stats = client.get_stats()
    print(f"✓ GET /stats: {stats}")
except Exception as e:
    print(f"✗ GET /stats: {e}")

try:
    cmd = client.get_command()
    print(f"✓ GET /get_command: {cmd}")
except Exception as e:
    print(f"✗ GET /get_command: {e}")

try:
    run = client.toggle_auto_run()
    print(f"✓ POST /toggle_auto_run: {run}")
    client.toggle_auto_run()  # toggle back
except Exception as e:
    print(f"✗ POST /toggle_auto_run: {e}")

try:
    res = client.send_command("Hello from antigravity-automation Python package test!")
    print(f"✓ POST /send_command: {res}")
except Exception as e:
    print(f"✗ POST /send_command: {e}")

try:
    res = client.track_action("remote_command")
    print(f"✓ POST /track_action: {res}")
except Exception as e:
    print(f"✗ POST /track_action: {e}")

print("\n✅ Live tests complete!")

File: libraries/npm_package/test/test_live.js
/**
 * Live integration test — requires the Antigravity Bridge running on localhost:5000
 * Run: npm run test:live
 */
const { AntigravityClient, AntigravityStream } = require('../dist');

async function main() {
    const client = new AntigravityClient();
    console.log('Testing against http://localhost:5000...\n');

    try {
        // Stats (GET — least destructive)
        const stats = await client.getStats();
        console.log('✓ GET /stats:', JSON.stringify(stats));
    } catch (e) {
        console.log('✗ GET /stats:', e.message);
    }

    try {
        // Get Command (GET)
        const cmd = await client.getCommand();
        console.log('✓ GET /get_command:', JSON.stringify(cmd));
    } catch (e) {
        console.log('✗ GET /get_command:', e.message);
    }

    try {
        // Toggle Auto-Run
        const run = await client.toggleAutoRun();
        console.log('✓ POST /toggle_auto_run:', JSON.stringify(run));
        // Toggle back
        await client.toggleAutoRun();
    } catch (e) {
        console.log('✗ POST /toggle_auto_run:', e.message);
    }

    try {
        // Send Command
        const cmd = await client.sendCommand('Hello from antigravity-automation npm package test!');
        console.log('✓ POST /send_command:', JSON.stringify(cmd));
    } catch (e) {
        console.log('✗ POST /send_command:', e.message);
    }

    try {
        // Track Action
        const track = await client.trackAction('remote_command');
        console.log('✓ POST /track_action:', JSON.stringify(track));
    } catch (e) {
        console.log('✗ POST /track_action:', e.message);
    }

    // WebSocket stream (5 second test)
    console.log('\nTesting WebSocket stream (5s)...');
    const stream = new AntigravityStream({ reconnect: false });
    let msgCount = 0;

    stream.onConnected(() => console.log('✓ WebSocket connected'));
    stream.onMessage((msg) => {
        msgCount++;
        console.log(`  Stream message #${msgCount}: [${msg.title}] ${msg.content.substring(0, 60)}...`);
    });
    stream.onError((err) => console.log('  WebSocket error:', err.message));

    stream.connect();

    await new Promise(resolve => setTimeout(resolve, 5000));
    stream.disconnect();
    console.log(`✓ WebSocket test done (${msgCount} messages received)`);

    console.log('\n✅ Live tests complete!');
}

main().catch(console.error);
